lib/routes/compact.js

```diff
@@ -7,11 +7,17 @@ module.exports = function (app) {
 
   // DB Compaction
   app.post('/:db/_compact', utils.jsonParser, function (req, res, next) {
-    req.db.compact(utils.makeOpts(req), function (err, response) {
-      if (err) {
-        return utils.sendError(res, err);
-      }
-      utils.sendJSON(res, 200, {ok: true});
+    // don't wait for compaction to finish, but do give errors a chance
+    // to pop up by waiting for a little bit. Yes, this is a hack.
+    var encounteredError = false;
+    req.db.compact(utils.makeOpts(req)).catch(function (err) {
+      encounteredError = true;
+      return utils.sendError(res, err);
     });
+    setTimeout(function () {
+      if (!encounteredError) {
+        utils.sendJSON(res, 202, {ok: true});
+      }
+    }, 0);
```

Comment:
Do you have to do this because CouchDB always returns immediately with ok: true? Why not just wait for compaction to fully finish before sending the response?